src/openrct2/ride/TrackDesignSave.cpp

```diff
@@ -1227,7 +1229,7 @@ bool track_design_save_to_file(const utf8 *path)
     auto_buffer_write(&td6Buffer, &EndMarker, sizeof(EndMarker));
 
     // Encode TD6 data
-    uint8 *encodedData = malloc(0x8000);
+    uint8 * encodedData = Memory::Allocate<uint8>(0x8000);
```

Comment:
If you change this, also change the deleter

Reply:
I wonder why nobody is using std::unique_ptr in such cases, you don't have to worry then.

Reply:
The Memory functions are deprecated, you should now use `std::unique_ptr` and `std::copy_n` and `std::fill_n`.